src/messaging/MessageCounterSync.cpp

```diff
@@ -76,11 +76,132 @@ void MessageCounterSyncMgr::OnMessageReceived(Messaging::ExchangeContext * excha
 
 void MessageCounterSyncMgr::OnResponseTimeout(Messaging::ExchangeContext * exchangeContext)
 {
+    mIsMsgCounterSyncReqInProgress = false;
+
     // Close the exchange if MsgCounterSyncRsp is not received before kMsgCounterSyncTimeout.
     if (exchangeContext != nullptr)
         exchangeContext->Close();
 }
 
+CHIP_ERROR MessageCounterSyncMgr::AddToRetransTable(uint16_t protocolId, uint8_t msgType, const SendFlags & sendFlags,
+                                                    System::PacketBufferHandle msgBuf, Messaging::ExchangeContext * exchangeContext)
+{
+    bool added     = false;
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    VerifyOrReturnError(exchangeContext != nullptr, err = CHIP_ERROR_INVALID_ARGUMENT);
+
+    for (RetransTableEntry & entry : mRetransTable)
+    {
+        // Check the exchContext pointer for finding an empty slot in Table
+        if (entry.exchangeContext == nullptr)
+        {
+            entry.protocolId      = protocolId;
+            entry.msgType         = msgType;
+            entry.msgBuf          = std::move(msgBuf);
+            entry.exchangeContext = exchangeContext;
+
+            // Increment the reference count
```

Comment:
nit: comment seems to just restate code. Could be removed or updated with something not as obvious (maybe `// Exchange reference counter will be released again when this entry is retransmitted`)

Reply:
Acked